Services/WebAccessChecker/classes/class.ilWebAccessChecker.php

```diff
@@ -212,6 +214,10 @@ public function initILIAS()
             }
         }
         $this->setInitialized(true);
+
+        if ($DIC->user()->getId() === 0) {
+            $DIC->user()->setId(ANONYMOUS_USER_ID);
```

Comment:
Hi @mjansenDatabay 
Thanks for the proposal! I'm currently not sure if one should set the users id manually to another, this may have some side-effects?

Reply:
Hi @chfsx ,

of course this is a workaround. The actual issue is: https://mantis.ilias.de/view.php?id=32063

There should not be a global user instance with ID 0, so this workaround simply sets the correct id for the context (user is not authenticated, so it must be an anonymous user). It is not a persistent change, only an 'in memory' state change of the users id. Possible (I think mostly possitive) side effects are correct permission checks etc..